### ISSUE TITLE
Intellisense not working when more than one DynamicParameter is used

### ISSUE THREAD

**Issue by DarkLite1** [PowerShell/vscode-powershell#2117]
Version: 1.37.0-insider (user setup)
Commit: 92da68a71cfb60bd3b9b0d7fbfb2a7e1fff9dbaf
Date: 2019-08-01T05:37:54.848Z
Electron: 4.2.7
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 6.2.9200

**Steps to Reproduce:**

Consider the following example function:

```powershell
Function Test-Stuff {
    Param (
        [Parameter(Mandatory, ValueFromPipeline)]
        [Object[]]$Ticket,
        [Switch]$PassThru
    )
    DynamicParam {
        $DynamicParameters = @(
            @{
                Name        = 'Environment'
                Type        = [Array]
                Position    = 0
                Mandatory   = $true
                ValidateSet = @('Test', 'Prod', 'Stage', 'Dev')
            }
            ,
            @{
                Name        = 'Type'
                Type        = [Array]
                Position    = 0
                Mandatory   = $true
                ValidateSet = @('Incident', 'Request', 'Project')
            }
        )
        $DynamicParameters | ForEach-Object { New-Object PSObject -Property $_ } | New-DynamicParameter
    }
    Begin {
        New-DynamicParameter -CreateVariables -BoundParameters $PSBoundParameters
    }
    Process {
        'yes'
    }
}

```
The function `New-DynamicParamter' can be downloaded [here](https://github.com/beatcracker/Powershell-Misc/blob/master/New-DynamicParameter.ps1).

- When using intellisense in the editor pane it keeps on loading and nothing is suggested for the second/last dynamic parameter (in this example `-Type`)

![image](https://user-images.githubusercontent.com/10828570/62292373-90c4ef00-b466-11e9-9e28-a6627b265b21.png)

- Observe that the same function with intellisense works fine in the `Terminal` or in the editor of the `PowerShell ISE`. 

![image](https://user-images.githubusercontent.com/10828570/62292505-d2ee3080-b466-11e9-845c-734c20ee7f85.png)

I was going nuts about this, because I thought my code was broken. Seems after a bit of stressing it's a bug in the editor pane. Thank you for looking at this.

**Comment by rjmholt** [comment]
@DarkLite1 do other IDE features continue to work after you hit this?

**Comment by SydneyhSmith** [comment]
@DarkLite1 thanks for reporting this...interestingly the editor and terminal are using the same API so it could be a performance reason as to why the parameters are not loading
It would be really helpful if you could also attach the logs--instructions for how to do that are [here ](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#providing-information-about-your-environment) and please use diagnostic logging `"powershell.developer.editorServicesLogLevel": "Diagnostic"`

**Comment by DarkLite1** [comment]
@rjmholt when this happens and I try another function to complete in the editor pane, like type `Get-Pr` hit `CTRL+I` and expect it to complete to `Get-Process`, it wont work anymore. When intellisense is invoked it keeps showing the text `Loading...` but nothing happens.
![image](https://user-images.githubusercontent.com/10828570/62602718-a87ff580-b8f4-11e9-88d3-95d1bec740db.png)

**Comment by DarkLite1** [comment]
@SydneyhSmith you can find the logs here:
[logs.zip](https://github.com/PowerShell/vscode-powershell/files/3475768/logs.zip)

Notice that the first run the function executed intelissense correctly, the second time I tried it it kept showing the text `Loading...` and nothing happened.

**Comment by DarkLite1** [comment]
This is strange... When I try it manually it works fine, like this:

```powershell
Function Test-Stuff {
    Param (
        [Parameter(Mandatory, ValueFromPipeline)]
        [Object[]]$Ticket,
        [Switch]$PassThru
    )
    DynamicParam {
        $EnvCustomAttribute = New-Object System.Management.Automation.ParameterAttribute
        $EnvCustomAttribute.Position = 0
        $EnvCustomAttribute.Mandatory = $true
        $EnvAttributeCollection = New-Object System.Collections.ObjectModel.Collection[System.Attribute]
        $EnvAttributeCollection.Add($EnvCustomAttribute)
        $EnvAttribValidateSet = New-Object System.Management.Automation.ValidateSetAttribute(@('Test', 'Prod', 'Stage', 'Dev'))
        $EnvAttributeCollection.Add($EnvAttribValidateSet)
        $EnvEnvironmentParam = New-Object System.Management.Automation.RuntimeDefinedParameter(
            'Environment', [String], $EnvAttributeCollection)

        $TypeCustomAttribute = New-Object System.Management.Automation.ParameterAttribute
        $TypeCustomAttribute.Position = 0
        $TypeCustomAttribute.Mandatory = $true
        $TypeAttributeCollection = New-Object System.Collections.ObjectModel.Collection[System.Attribute]
        $TypeAttributeCollection.Add($TypeCustomAttribute)
        $TypeAttribValidateSet = New-Object System.Management.Automation.ValidateSetAttribute(@('Incident', 'Request', 'Project'))
        $TypeAttributeCollection.Add($TypeAttribValidateSet)
        $TypeEnvironmentParam = New-Object System.Management.Automation.RuntimeDefinedParameter(
            'Type', [String], $TypeAttributeCollection)

        $ParamDictionary = New-Object System.Management.Automation.RuntimeDefinedParameterDictionary
        $ParamDictionary.Add('Environment', $EnvEnvironmentParam)
        $ParamDictionary.Add('Type', $TypeEnvironmentParam)
        $ParamDictionary
    }

    Process {
        'yes'
    }
}

```

But when I use the example code from my first post with the function it doesn't work as desired in vscode, but it does in PowerShell ISE. So somewhere there's a difference in the dynamic parameters that make intellisense crash.

**Comment by DarkLite1** [comment]
I made an export of both dynamic parameters, one that is generated by the function and one that is made manually. This will allow someone with more knowledge to see what the difference in them is and why intellisense doesn't work in `vscode` but does in the `PowerShell ISE`.

[DynamicParams.zip](https://github.com/PowerShell/vscode-powershell/files/3475865/DynamicParams.zip)

**Comment by rjmholt** [comment]
@DarkLite1 based on your response, I suspect you're hitting https://github.com/PowerShell/PowerShellEditorServices/issues/762#issuecomment-428756484. @SeeminglyScience might have some insight there.

This is something we don't have much control over; the fix lies in PackageManagement, but we're investigating how to fix it since it also affects PSScriptAnalyzer

**Comment by SeeminglyScience** [comment]
> @DarkLite1 based on your response, I suspect you're hitting [PowerShell/PowerShellEditorServices#762 (comment)](https://github.com/PowerShell/PowerShellEditorServices/issues/762#issuecomment-428756484). @SeeminglyScience might have some insight there.

Yeah the OP doesn't hang for me, so there's a good chance it's related.

**Comment by TylerLeonhardt** [comment]
Bot is trigger happy...
